prophic3/prophic3.cpp

```diff
@@ -1014,9 +993,26 @@ void ProphIC3::print_witness(msat_model model,
   logger(1) << endlog;
   logger(1) << "+++++++++++++++++++++ UF values ++++++++++++++++++++"
             << endlog;
-  for (auto elem : abstractor.read_ufs()) {
-    msat_term arr = elem.first;
-    msat_decl fun = elem.second;
+
+  // collect all the arrays
+  TermList abs_arrays;
+  for (auto sv : abs_ts_.statevars()) {
+    msat_term concsv = abstractor.concrete(sv);
```

Comment:
can you use get_orig_type here?

Reply:
Yep! Good catch. Changed with: 43a7b4464767d6baf8d5cbe4cfb8b307645ba56c